### ISSUE TITLE
Update fibermap to match targeting inputs

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#679]
Related to desihub/desidatamodel#41 (datamodel consistency), but opening a self-assigned desispec-specific issue to track this: update fibermap to use a structure that matches input targeting, and fix code like flux calibration that will break as a result (e.g. MAG[i] -> FLUX_G/R/Z)

**Comment by sbailey** [comment]
Changes implemented in PR #717 for consistency with target catalogs and fiber assignment:

## Unchanged columns
```
TARGETID, DESI_TARGET, BGS_TARGET, MWS_TARGET
FIBER, PETAL_LOC, DEVICE_LOC, LOCATION
LAMBDAREF
SPECTROID
BRICKNAME
```

## Columns that changed name or meaning
```
Old                     ->  New
OBJTYPE (ELG, LRG, ...)     OBJTYPE  (TGT, SKY, BAD)
RA_TARGET, DEC_TARGET       TARGET_RA, TARGET_DEC
MAG, FILTER                 FLUX_G, FLUX_R, FLUX_Z, FLUX_W1, FLUX_W2
RA_OBS, DEC_OBS             FIBER_RA, FIBER_DEC
X_TARGET, Y_TARGET          DESIGN_X, DESIGN_Y
X_FVCOBS, Y_FVCOBS          DELTA_X, DELTA_Y (wrt intended position)
                            --> should have been DELTA_XFPA
X_FVCERR, Y_FVCERR          DELTA_X_IVAR, DELTA_Y_IVAR
POSITIONER                  DEVICE_LOC
TARGETCAT                   (dropped)
```

## New Columns propagated from tractor catalogs
```
BRICKID (int)
TARGET_RA_IVAR, TARGET_DEC_IVAR  (originally RA_IVAR, DEC_IVAR)
BRICK_OBJID (originally OBJID from Tractor)
MORPHTYPE (originally TYPE from Tractor)
REF_ID, PMRA, PMDEC, PMRA_IVAR, PMDEC_IVAR (proper motion info)
FLUX_IVAR_[G,R,Z,W1,W2]  (flux inverse variances)
FIBERFLUX_[G,R,Z,W1,W2]  (model flux in fiber)
FIBERTOTFLUX_[G,R,Z,W1,W2]  (model+overlapping models flux in fiber)
MWTRANSMISSION_[G,R,Z] (Milky Way dust transmission, 0-1)
EBV (galactic dust reddening E(B-V))
PHOTSYS (N/S)
```

## New Columns propagated from target selection / MTL
```
SECONDARY_TARGET  (targeting bits, not yet implemented)
PRIORITY (int), SUBPRIORITY (float 0-1)
```

## New columns propagated from fiber assignment
```
DESIGN_Q, DESIGN_S (radial coordinate system on focal plane)
FIBERSTATUS (bitmask of known fiber problems 0=ok)
LAMBDA_REF (oops; also LAMBDAREF)
NUMTARGET (number of targets covered by this positioner)
```

## New columns added by ICS / PlateMaker
```
NUM_ITER (number of movement iterations needed by PlateMaker / FVC)
FIBER_RA_IVAR, FIBER_DEC_IVAR
```

## Changes from the format used at Mock Observing June 2018

These need to be reviewed for whether they are mistakes or ok:

### Dropped
```
COMM_TARGET, SV1_DESI_TARGET, SV1_BGS_TARGET, SV1_MWS_TARGET
DELTA_XFPA vs. DELTA_X etc.
```

### Added since "Mock Observing Format"
BRICKNAME (maybe want to re-drop)
DELTA_X vs. DELTA_XFPA etc.
EBV
FIBERFLUX_W1, FIBERFLUX_W2
FIBERTOTFLUX_W1, FIBERTOTFLUX_W2
LAMBDAREF
MW_TRANSMISSION_G, MW_TRANSMISSION_R, MW_TRANSMISSION_Z
PHOTSYS

## Other known changes to make in the future
* Drop `BRICKNAME` (use `BRICKID` only)
* `DELTA_X` -> `DELTA_XFPA`
* `LAMBDAREF` vs. `LAMBDA_REF`
*  Propagate *all* target selection columns into fibermap instead of separate table

**Comment by sbailey** [comment]
implemented in PR #717
